File: 1-Two-Sum/solution.java
/*
1. hashmap : key : nums[i],  value: key
2. a + b = tar => a = tar - b

*/

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                res[0] = map.get(target - nums[i]);
                res[1] = i;
                break;
            }
            else {
                map.put(nums[i], i);
            }
        }
        return res;
    }
}

File: 1-Two-Sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        res = []
        map = {}
        for i in xrange(len(nums)):
            if target - nums[i] in map:
                res.append(map[target - nums[i]])
                res.append(i)
                break
            else:
                map[nums[i]] = i
        return res